pkg/apis/wksprovider/controller/wksctl/machine_actuator.go

```diff
@@ -79,7 +110,16 @@ func (a *MachineActuator) create(ctx context.Context, cluster *clusterv1.Cluster
 	if err = a.initializeMasterPlanIfNecessary(installer); err != nil {
 		return err
 	}
-	nodePlan, err := a.getNodePlan(c, machine, m.Private.Address, installer)
+	// Also, update footloose IP from env
+	log.Infof("FETCHING FOOTLOOSE ADDRESS...")
+	fip := goos.Getenv("FOOTLOOSE_SERVER_ADDR")
+	if fip != "" {
+		footlooseAddr = fip
+		footlooseBackend = goos.Getenv("FOOTLOOSE_BACKEND")
```

Comment:
We should check to see if we get a value back and use the default if it isn't set.